diff -crBN ACE_wrappers/ace/ace.mwc ACE_wrappers.TunnelEx/ace/ace.mwc
*** ACE_wrappers/ace/ace.mwc	2010-09-15 16:42:34.000000000 +0700
--- ACE_wrappers.TunnelEx/ace/ace.mwc	2011-05-13 00:12:18.319579400 +0700
***************
*** 2,6 ****
--- 2,8 ----
  // $Id: ace.mwc 91777 2010-09-15 09:42:34Z johnnyw $
  
  workspace {
+   ace.mpc
+   SSL/ssl.mpc
  }
  
diff -crBN ACE_wrappers/ace/config-win32-common.h ACE_wrappers.TunnelEx/ace/config-win32-common.h
*** ACE_wrappers/ace/config-win32-common.h	2010-10-01 19:00:01.000000000 +0700
--- ACE_wrappers.TunnelEx/ace/config-win32-common.h	2011-05-13 00:12:18.377582700 +0700
***************
*** 619,625 ****
  # endif /* !ACE_HAS_WINCE */
  
  #if !defined (WINVER)
! # define WINVER 0x0400 // pretend it's at least WinNT 4.0
  #endif
  
  ///////////////////////////////////////
--- 619,625 ----
  # endif /* !ACE_HAS_WINCE */
  
  #if !defined (WINVER)
! # define WINVER 0x0501 // pretend it's at least WinNT 4.0
  #endif
  
  ///////////////////////////////////////
diff -crBN ACE_wrappers/ace/config.h ACE_wrappers.TunnelEx/ace/config.h
*** ACE_wrappers/ace/config.h	1970-01-01 06:00:00.000000000 +0600
--- ACE_wrappers.TunnelEx/ace/config.h	2011-05-13 00:12:18.385583100 +0700
***************
*** 0 ****
--- 1,18 ----
+ /**************************************************************************
+  *   Created: 2010/06/11 09:00
+  *    Author: Eugene V. Palchukovsky
+  *    E-mail: eugene@palchukovsky.com
+  * -------------------------------------------------------------------
+  *   Project: TunnelEx
+  *       URL: http://tunnelex.net
+  **************************************************************************/
+ 
+ #ifndef INCLUDED_FILE__TUNNELEX__config_h__1005091822
+ #define INCLUDED_FILE__TUNNELEX__config_h__1005091822
+ 
+ #define ACE_HAS_ICMP_SUPPORT 1
+ #define ACE_USES_WCHAR 1
+ 
+ #include <ace/config-win32.h>
+ 
+ #endif // INCLUDED_FILE__TUNNELEX__config_h__1005091822
diff -crBN ACE_wrappers/ace/Recursive_Thread_Mutex.cpp ACE_wrappers.TunnelEx/ace/Recursive_Thread_Mutex.cpp
*** ACE_wrappers/ace/Recursive_Thread_Mutex.cpp	2010-09-09 18:21:50.000000000 +0700
--- ACE_wrappers.TunnelEx/ace/Recursive_Thread_Mutex.cpp	2011-05-13 00:57:46.940647600 +0700
***************
*** 61,72 ****
  {
    // ACE_TRACE ("ACE_Recursive_Thread_Mutex::get_thread_id");
  #if defined (ACE_HAS_RECURSIVE_MUTEXES)
!   // @@ The structure CriticalSection in Win32 doesn't hold the thread
!   // handle of the thread that owns the lock.  However it is still not
!   // clear at this point how to translate a thread handle to its
!   // corresponding thread id.
!   errno = ENOTSUP;
!   return ACE_OS::NULL_thread;
  #else
    ACE_thread_t owner_id;
    ACE_OS::mutex_lock (&this->lock_.nesting_mutex_);
--- 61,67 ----
  {
    // ACE_TRACE ("ACE_Recursive_Thread_Mutex::get_thread_id");
  #if defined (ACE_HAS_RECURSIVE_MUTEXES)
!   return ACE_thread_t(this->lock_.OwningThread);
  #else
    ACE_thread_t owner_id;
    ACE_OS::mutex_lock (&this->lock_.nesting_mutex_);
diff -crBN ACE_wrappers/ace/WIN32_Asynch_IO.cpp ACE_wrappers.TunnelEx/ace/WIN32_Asynch_IO.cpp
*** ACE_wrappers/ace/WIN32_Asynch_IO.cpp	2010-08-16 20:03:34.000000000 +0700
--- ACE_wrappers.TunnelEx/ace/WIN32_Asynch_IO.cpp	2011-05-13 00:12:18.393583600 +0700
***************
*** 1713,1718 ****
--- 1713,1764 ----
    return return_val;
  }
  
+ // copied from ACE_WIN32_Asynch_Write_Stream::shared_write
+ // see TEX-653 for details
+ int
+ ACE_WIN32_Asynch_Write_File::shared_write (ACE_WIN32_Asynch_Write_Stream_Result *result)
+ {
+   u_long bytes_written;
+   if (result->bytes_to_write () > MAXDWORD)
+     {
+       errno = ERANGE;
+       return -1;
+     }
+   DWORD bytes_to_write = static_cast<DWORD> (result->bytes_to_write ());
+ 
+   result->set_error (0); // Clear error before starting IO.
+ 
+   int initiate_result = 0;
+   initiate_result = ::WriteFile (result->handle (),
+                                  result->message_block ().rd_ptr (),
+                                  bytes_to_write,
+                                  &bytes_written,
+                                  result);
+   if (initiate_result == 1)
+     // Immediate success: the OVERLAPPED will still get queued.
+     return 0;
+ 
+   // If initiate failed, check for a bad error.
+   ACE_OS::set_errno_to_last_error ();
+   switch (errno)
+     {
+     case ERROR_IO_PENDING:
+       // The IO will complete proactively: the OVERLAPPED will still
+       // get queued.
+       return 0;
+ 
+     default:
+       // Something else went wrong: the OVERLAPPED will not get
+       // queued.
+ 
+       if (ACE::debug ())
+         ACE_DEBUG ((LM_ERROR,
+                     ACE_TEXT ("%p\n"),
+                     ACE_TEXT ("Initiating write")));
+       return -1;
+     }
+ }
+ 
  int
  ACE_WIN32_Asynch_Write_File::writev (ACE_Message_Block &message_block,
                                       size_t bytes_to_write,
diff -crBN ACE_wrappers/ace/WIN32_Asynch_IO.h ACE_wrappers.TunnelEx/ace/WIN32_Asynch_IO.h
*** ACE_wrappers/ace/WIN32_Asynch_IO.h	2010-10-21 18:15:17.000000000 +0700
--- ACE_wrappers.TunnelEx/ace/WIN32_Asynch_IO.h	2011-05-13 00:12:18.400584000 +0700
***************
*** 960,965 ****
--- 960,972 ----
                const void *act,
                int priority,
                int signal_number = 0);
+ 
+ protected:
+ 
+ 	// copied from ACE_WIN32_Asynch_Write_Stream::shared_write
+ 	// see TEX-653 for details
+ 	int shared_write (ACE_WIN32_Asynch_Write_Stream_Result *result);
+ 
  };
  
  /**
diff -crBN ACE_wrappers/bin/MakeProjectCreator/config/ace_output.mpb ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/ace_output.mpb
*** ACE_wrappers/bin/MakeProjectCreator/config/ace_output.mpb	2008-03-04 20:51:23.000000000 +0600
--- ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/ace_output.mpb	2011-05-13 00:12:18.408584500 +0700
***************
*** 2,8 ****
  // $Id: ace_output.mpb 80826 2008-03-04 14:51:23Z wotte $
  
  project {
!   libout = $(ACE_ROOT)/lib
  
    specific(automake) {
      libout -= $(ACE_ROOT)/lib
--- 2,9 ----
  // $Id: ace_output.mpb 80826 2008-03-04 14:51:23Z wotte $
  
  project {
!   dllout = $(BinDir)
!   libout = $(LibDir)
  
    specific(automake) {
      libout -= $(ACE_ROOT)/lib
diff -crBN ACE_wrappers/bin/MakeProjectCreator/config/acedefaults.mpb ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/acedefaults.mpb
*** ACE_wrappers/bin/MakeProjectCreator/config/acedefaults.mpb	2010-06-23 10:24:27.000000000 +0700
--- ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/acedefaults.mpb	2011-05-13 00:12:18.419585100 +0700
***************
*** 4,10 ****
  project: ipv6, vc_warnings, build_files, test_files, svc_conf_files, ace_unicode {
    staticflags += ACE_AS_STATIC_LIBS
    includes    += $(ACE_ROOT)
!   libpaths    += $(ACE_ROOT)/lib
  
    // Support the alternative Borland Make project type
    specific(bmake) {
--- 4,10 ----
  project: ipv6, vc_warnings, build_files, test_files, svc_conf_files, ace_unicode {
    staticflags += ACE_AS_STATIC_LIBS
    includes    += $(ACE_ROOT)
!   libpaths    += $(LibDir)
  
    // Support the alternative Borland Make project type
    specific(bmake) {
diff -crBN ACE_wrappers/bin/MakeProjectCreator/config/default.rel ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/default.rel
*** ACE_wrappers/bin/MakeProjectCreator/config/default.rel	2008-03-04 20:51:23.000000000 +0600
--- ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/default.rel	2011-05-13 00:12:18.436586100 +0700
***************
*** 4,8 ****
--- 4,9 ----
  // The second (optional) column is the value to build up if it isn't defined
  // as an environmenment variable.
  *_ROOT
+ *Dir
  TAO_ROOT,  $ACE_ROOT/TAO
  CIAO_ROOT, $TAO_ROOT/CIAO
diff -crBN ACE_wrappers/bin/MakeProjectCreator/config/global.features ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/global.features
*** ACE_wrappers/bin/MakeProjectCreator/config/global.features	2010-10-20 16:26:46.000000000 +0700
--- ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/global.features	2011-05-13 00:12:18.451586900 +0700
***************
*** 18,31 ****
  // settings can be set here.
  
  // Enable these by default.
! acexml        = 1
  ace_svcconf   = 1
! ace_token     = 1
  
  // By default we disable these.
  ace_for_tao   = 0
  cross_compile = 0
! ssl           = 0
  qos           = 0
  rapi          = 0
  repo          = 0
--- 18,31 ----
  // settings can be set here.
  
  // Enable these by default.
! acexml        = 0
  ace_svcconf   = 1
! ace_token     = 0
  
  // By default we disable these.
  ace_for_tao   = 0
  cross_compile = 0
! ssl           = 1
  qos           = 0
  rapi          = 0
  repo          = 0
***************
*** 89,95 ****
  // 1 - Athena widgets libraries are available
  athena        = 0
  // 1 - X11 libraries are available
! x11           = 1
  // 1 - X Toolkit libraries are available
  xt            = 0
  // 1 - Tk libraries are available
--- 89,95 ----
  // 1 - Athena widgets libraries are available
  athena        = 0
  // 1 - X11 libraries are available
! x11           = 0
  // 1 - X Toolkit libraries are available
  xt            = 0
  // 1 - Tk libraries are available
***************
*** 104,134 ****
  // 1 - Fox libraries are available
  fox           = 0
  // 1 - OpenGL libraries are available
! gl            = 1
  
  // Features for ACE support for various GUI libraries
  // Use QtReactor
! ace_qtreactor    = 1
  // Use XtReactor
! ace_xtreactor    = 1
  // Use TkReactor
! ace_tkreactor    = 1
  // Use FlReactor
! ace_flreactor    = 1
  // Use FoxReactor
! ace_foxreactor   = 1
  
  // Features for TAO support for various GUI libraries
  // Use QtResources for TAO
! tao_qtresource    = 1
  // Use XtResources for TAO
! tao_xtresource    = 1
  // Use TkResources for TAO
! tao_tkresource    = 1
  // Use FlResources for TAO
! tao_flresource    = 1
  // Use FoxResources for TAO
! tao_foxresource    = 1
  
  //Use the ZIP_Wrapper feature for when using the ZIP wrapper classes
  ZIP_Wrapper = 0
--- 104,134 ----
  // 1 - Fox libraries are available
  fox           = 0
  // 1 - OpenGL libraries are available
! gl            = 0
  
  // Features for ACE support for various GUI libraries
  // Use QtReactor
! ace_qtreactor    = 0
  // Use XtReactor
! ace_xtreactor    = 0
  // Use TkReactor
! ace_tkreactor    = 0
  // Use FlReactor
! ace_flreactor    = 0
  // Use FoxReactor
! ace_foxreactor   = 0
  
  // Features for TAO support for various GUI libraries
  // Use QtResources for TAO
! tao_qtresource    = 0
  // Use XtResources for TAO
! tao_xtresource    = 0
  // Use TkResources for TAO
! tao_tkresource    = 0
  // Use FlResources for TAO
! tao_flresource    = 0
  // Use FoxResources for TAO
! tao_foxresource    = 0
  
  //Use the ZIP_Wrapper feature for when using the ZIP wrapper classes
  ZIP_Wrapper = 0
diff -crBN ACE_wrappers/bin/MakeProjectCreator/config/vc10.features ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/vc10.features
*** ACE_wrappers/bin/MakeProjectCreator/config/vc10.features	2009-09-17 01:45:37.000000000 +0700
--- ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/vc10.features	2011-05-13 00:12:18.465587700 +0700
***************
*** 1,6 ****
  // $Id: vc10.features 86721 2009-09-16 18:45:37Z johnnyw $
  
! ssl=0
  qos=1
  rwho=0
  sctp=0
--- 1,6 ----
  // $Id: vc10.features 86721 2009-09-16 18:45:37Z johnnyw $
  
! ssl=1
  qos=1
  rwho=0
  sctp=0
diff -crBN ACE_wrappers/bin/MakeProjectCreator/config/vc8.features ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/vc8.features
*** ACE_wrappers/bin/MakeProjectCreator/config/vc8.features	2009-09-17 01:45:37.000000000 +0700
--- ACE_wrappers.TunnelEx/bin/MakeProjectCreator/config/vc8.features	2011-05-13 00:12:18.478588500 +0700
***************
*** 1,6 ****
  // $Id: vc8.features 86721 2009-09-16 18:45:37Z johnnyw $
  
! ssl=0
  qos=1
  rwho=0
  sctp=0
--- 1,6 ----
  // $Id: vc8.features 86721 2009-09-16 18:45:37Z johnnyw $
  
! ssl=1
  qos=1
  rwho=0
  sctp=0
diff -crBN ACE_wrappers/MPC/config/default.rel ACE_wrappers.TunnelEx/MPC/config/default.rel
*** ACE_wrappers/MPC/config/default.rel	2009-09-03 19:57:26.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/config/default.rel	2011-05-13 00:12:18.496589500 +0700
***************
*** 4,6 ****
--- 4,7 ----
  // The second (optional) column, which must be comma separated, is the value
  // to build up if it isn't defined as an environmenment variable.
  *_ROOT
+ *Dir
\ No newline at end of file
diff -crBN ACE_wrappers/MPC/config/openssl.mpb ACE_wrappers.TunnelEx/MPC/config/openssl.mpb
*** ACE_wrappers/MPC/config/openssl.mpb	2009-05-07 22:44:53.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/config/openssl.mpb	2011-05-13 00:12:18.513590500 +0700
***************
*** 6,18 ****
  // default. If we ever add a new ssl library, then you
  // would likely enable only one ssl library feature.
  feature(openssl) {
!   includes += $(SSL_ROOT)/include
!   libpaths += $(SSL_ROOT)/lib
  
    specific(prop:windows) {
      lit_libs += libeay32 ssleay32
-     includes += $(SSL_ROOT)/inc32
-     libpaths += $(SSL_ROOT)/out32dll $(SSL_ROOT)/out32
    } else {
      lit_libs += ssl crypto
  
--- 6,15 ----
  // default. If we ever add a new ssl library, then you
  // would likely enable only one ssl library feature.
  feature(openssl) {
!   includes += $(OpenSslDir)/$(OpenSslVersion)
  
    specific(prop:windows) {
      lit_libs += libeay32 ssleay32
    } else {
      lit_libs += ssl crypto
  
***************
*** 27,32 ****
      libpaths += $(SSL_ROOT)/lib/Builder5
    }
    specific(prop:microsoft) {
!     libpaths += $(SSL_ROOT)/lib/VC
    }
  }
--- 24,29 ----
      libpaths += $(SSL_ROOT)/lib/Builder5
    }
    specific(prop:microsoft) {
! 
    }
  }
diff -crBN ACE_wrappers/MPC/templates/vc10.mpd ACE_wrappers.TunnelEx/MPC/templates/vc10.mpd
*** ACE_wrappers/MPC/templates/vc10.mpd	2010-11-02 00:36:39.000000000 +0600
--- ACE_wrappers.TunnelEx/MPC/templates/vc10.mpd	2011-05-13 00:12:18.546592400 +0700
***************
*** 305,311 ****
  <%if(TreatSpecificWarningsAsErrors)%>
        <TreatSpecificWarningsAsErrors><%TreatSpecificWarningsAsErrors%>;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
  <%endif%>
! <%if(WarnAsError)%>
        <TreatWarningAsError>true</TreatWarningAsError>
  <%endif%>
  <%if(CompileAsManaged)%>
--- 305,311 ----
  <%if(TreatSpecificWarningsAsErrors)%>
        <TreatSpecificWarningsAsErrors><%TreatSpecificWarningsAsErrors%>;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
  <%endif%>
! <%if(WarnAsError && compares(WarnAsError, true))%>
        <TreatWarningAsError>true</TreatWarningAsError>
  <%endif%>
  <%if(CompileAsManaged)%>
diff -crBN ACE_wrappers/MPC/templates/vc10dll.mpt ACE_wrappers.TunnelEx/MPC/templates/vc10dll.mpt
*** ACE_wrappers/MPC/templates/vc10dll.mpt	2010-08-18 05:03:31.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/templates/vc10dll.mpt	2011-05-13 00:12:18.574594000 +0700
***************
*** 4,32 ****
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Release
  common_defines     = WIN32 _WINDOWS _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 2
! MinimalRebuild     = false
! GenerateDebugInformation = true
  
  conditional_include "vc10platforms"
  
  Release {
    optimize         = MaxSpeed
!   defines          = NDEBUG
    output_dir       = Release
!   intermediate_dir = Release
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG
!   output_dir       = .
!   intermediate_dir = Debug
!   lib_modifier     = d
  }
  
  Memcheck Release {
--- 4,52 ----
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Test Release
  common_defines     = WIN32 _WINDOWS _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 2
! 
! ExceptionHandling  = Async
! WarnAsError        = true
! OptimizeForWindows98 = 1
! warning_level      = 4
  
  conditional_include "vc10platforms"
  
  Release {
    optimize         = MaxSpeed
!   defines          = NDEBUG;BOOST_DISABLE_ASSERTS
    output_dir       = Release
!   intermediate_dir = $(IntDir)\Release
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
+   MinimalRebuild   = false
+   GenerateDebugInformation = false
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Debug
!   intermediate_dir = $(IntDir)\Debug
!   lib_modifier     = _dbg
!   LinkIncremental  = 2
!   MinimalRebuild   = true
!   GenerateDebugInformation = true
! }
! 
! Test {
!   optimize         = MaxSpeed
!   defines          = TEST;NDEBUG;PROFILE;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Test
!   intermediate_dir = $(IntDir)\Test
!   lib_modifier     = _test
!   EnableIntrinsicFunctions = true
!   WholeProgramOptimization = true
!   MinimalRebuild   = false
!   GenerateDebugInformation = true
  }
  
  Memcheck Release {
diff -crBN ACE_wrappers/MPC/templates/vc10exe.mpt ACE_wrappers.TunnelEx/MPC/templates/vc10exe.mpt
*** ACE_wrappers/MPC/templates/vc10exe.mpt	2010-08-18 05:03:31.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/templates/vc10exe.mpt	2011-05-13 00:12:18.588594800 +0700
***************
*** 4,33 ****
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Release
  common_defines     = WIN32 _CONSOLE _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 1
  SubSystem          = Console
! MinimalRebuild     = false
! GenerateDebugInformation = true
  
  conditional_include "vc10platforms"
  
  Release {
-   LinkIncremental  = 1
    optimize         = MaxSpeed
!   defines          = NDEBUG
    output_dir       = Release
!   intermediate_dir = Release
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG
!   intermediate_dir = Debug
!   lib_modifier     = d
  }
  
  Memcheck Release {
--- 4,53 ----
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Test Release
  common_defines     = WIN32 _CONSOLE _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 1
  SubSystem          = Console
! 
! ExceptionHandling  = Async
! WarnAsError        = true
! OptimizeForWindows98 = 1
! warning_level      = 4
  
  conditional_include "vc10platforms"
  
  Release {
    optimize         = MaxSpeed
!   defines          = NDEBUG;BOOST_DISABLE_ASSERTS
    output_dir       = Release
!   intermediate_dir = $(IntDir)\Release
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
+   MinimalRebuild   = false
+   GenerateDebugInformation = false
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Debug
!   intermediate_dir = $(IntDir)\Debug
!   lib_modifier     = _dbg
!   LinkIncremental  = 2
!   MinimalRebuild   = true
!   GenerateDebugInformation = true
! }
! 
! Test {
!   optimize         = MaxSpeed
!   defines          = TEST;NDEBUG;PROFILE;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Test
!   intermediate_dir = $(IntDir)\Test
!   lib_modifier     = _test
!   EnableIntrinsicFunctions = true
!   WholeProgramOptimization = true
!   MinimalRebuild   = false
!   GenerateDebugInformation = true
  }
  
  Memcheck Release {
diff -crBN ACE_wrappers/MPC/templates/vc10lib.mpt ACE_wrappers.TunnelEx/MPC/templates/vc10lib.mpt
*** ACE_wrappers/MPC/templates/vc10lib.mpt	2010-08-18 05:03:31.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/templates/vc10lib.mpt	2011-05-13 00:12:18.600595400 +0700
***************
*** 4,33 ****
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Release
  common_defines     = WIN32 _WINDOWS _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 4
! MinimalRebuild     = false
! GenerateDebugInformation = true
  
  conditional_include "vc10platforms"
  
  Release {
    optimize         = MaxSpeed
!   defines          = NDEBUG
!   output_dir       = Static_Release
!   intermediate_dir = Static_Release
!   lib_modifier     = s
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG
!   output_dir       = Static_Debug
!   intermediate_dir = Static_Debug
!   lib_modifier     = sd
  }
  
  Memcheck Release {
--- 4,52 ----
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Test Release
  common_defines     = WIN32 _WINDOWS _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 4
! 
! ExceptionHandling  = Async
! WarnAsError        = true
! OptimizeForWindows98 = 1
! warning_level      = 4
  
  conditional_include "vc10platforms"
  
  Release {
    optimize         = MaxSpeed
!   defines          = NDEBUG;BOOST_DISABLE_ASSERTS
!   output_dir       = Release
!   intermediate_dir = $(IntDir)\Release
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
+   MinimalRebuild   = false
+   GenerateDebugInformation = false
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Debug
!   intermediate_dir = $(IntDir)\Debug
!   lib_modifier     = _dbg
!   LinkIncremental  = 2
!   MinimalRebuild   = true
!   GenerateDebugInformation = true
! }
! 
! Test {
!   optimize         = MaxSpeed
!   defines          = TEST;NDEBUG;PROFILE;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Test
!   intermediate_dir = $(IntDir)\Test
!   lib_modifier     = _test
!   EnableIntrinsicFunctions = true
!   WholeProgramOptimization = true
!   MinimalRebuild   = false
!   GenerateDebugInformation = true
  }
  
  Memcheck Release {
diff -crBN ACE_wrappers/MPC/templates/vc10libexe.mpt ACE_wrappers.TunnelEx/MPC/templates/vc10libexe.mpt
*** ACE_wrappers/MPC/templates/vc10libexe.mpt	2010-08-18 05:03:31.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/templates/vc10libexe.mpt	2011-05-13 00:12:18.612596100 +0700
***************
*** 4,35 ****
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Release
  common_defines     = WIN32 _CONSOLE _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 1
  SubSystem          = Console
! MinimalRebuild     = false
! GenerateDebugInformation = true
  
  conditional_include "vc10platforms"
  
  Release {
-   LinkIncremental  = 1
    optimize         = MaxSpeed
!   defines          = NDEBUG
!   output_dir       = Static_Release
!   intermediate_dir = Static_Release
!   lib_modifier     = s
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG
!   output_dir       = Static_Debug
!   intermediate_dir = Static_Debug
!   lib_modifier     = sd
  }
  
  Memcheck Release {
--- 4,53 ----
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Test Release
  common_defines     = WIN32 _CONSOLE _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 1
  SubSystem          = Console
! 
! ExceptionHandling  = Async
! WarnAsError        = true
! OptimizeForWindows98 = 1
! warning_level      = 4
  
  conditional_include "vc10platforms"
  
  Release {
    optimize         = MaxSpeed
!   defines          = NDEBUG;BOOST_DISABLE_ASSERTS
!   output_dir       = Release
!   intermediate_dir = $(IntDir)\Release
    EnableIntrinsicFunctions = true
    WholeProgramOptimization = true
+   MinimalRebuild   = false
+   GenerateDebugInformation = false
  }
  
  Debug {
    debug_prj        = 1
!   defines          = _DEBUG;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Debug
!   intermediate_dir = $(IntDir)\Debug
!   lib_modifier     = _dbg
!   LinkIncremental  = 2
!   MinimalRebuild   = true
!   GenerateDebugInformation = true
! }
! 
! Test {
!   optimize         = MaxSpeed
!   defines          = TEST;NDEBUG;PROFILE;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Test
!   intermediate_dir = $(IntDir)\Test
!   lib_modifier     = _test
!   EnableIntrinsicFunctions = true
!   WholeProgramOptimization = true
!   MinimalRebuild   = false
!   GenerateDebugInformation = true
  }
  
  Memcheck Release {
diff -crBN ACE_wrappers/MPC/templates/vc8dll.mpt ACE_wrappers.TunnelEx/MPC/templates/vc8dll.mpt
*** ACE_wrappers/MPC/templates/vc8dll.mpt	2010-04-11 02:38:14.000000000 +0700
--- ACE_wrappers.TunnelEx/MPC/templates/vc8dll.mpt	2011-05-13 00:12:18.619596500 +0700
***************
*** 4,32 ****
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Release
  common_defines     = WIN32 _WINDOWS _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 2
  Detect64BitPortabilityProblems = false
! MinimalRebuild     = false
  
  conditional_include "vc8platforms"
  
  Release {
    LinkIncremental  = 1
    optimize         = 2
!   defines          = NDEBUG
    output_dir       = Release
!   intermediate_dir = Release
    GenerateDebugInformation = false
  }
  
  Debug {
!   debug_prj        = 1
!   defines          = _DEBUG
!   output_dir       = .
!   intermediate_dir = Debug
!   lib_modifier     = d
  }
  
  Memcheck Release {
--- 4,52 ----
  conditional_include "common"
  conditional_include "windowscommon"
  
! configurations     = Debug Test Release
  common_defines     = WIN32 _WINDOWS _CRT_NONSTDC_NO_WARNINGS
  ConfigurationType  = 2
  Detect64BitPortabilityProblems = false
! WarnAsError        = true
! ExceptionHandling  = 2
! OptimizeForWindows98 = 1
! warning_level      = 4
  
  conditional_include "vc8platforms"
  
  Release {
    LinkIncremental  = 1
    optimize         = 2
!   defines          = NDEBUG;BOOST_DISABLE_ASSERTS
    output_dir       = Release
!   intermediate_dir = $(IntDir)\Release
    GenerateDebugInformation = false
+   WholeProgramOptimization = 1
+   MinimalRebuild   = false
+ }
+ 
+ Test {
+   LinkIncremental  = 2
+   optimize         = 2
+   defines          = TEST;NDEBUG;PROFILE;BOOST_ENABLE_ASSERT_HANDLER
+   output_dir       = Test
+   intermediate_dir = $(IntDir)\Test
+   lib_modifier     = _test
+   WholeProgramOptimization = 1
+   MinimalRebuild   = false
+   GenerateDebugInformation = true
  }
  
  Debug {
!   debug_prj        = 2
!   defines          = _DEBUG;BOOST_ENABLE_ASSERT_HANDLER
!   output_dir       = Debug
!   intermediate_dir = $(IntDir)\Debug
!   lib_modifier     = _dbg
!   debug_format     = 4
!   MinimalRebuild   = true
!   GenerateDebugInformation = true
  }
  
  Memcheck Release {
